From 91547b21cbbdc608a800ad53f47f296798b370ab Mon Sep 17 00:00:00 2001
From: Samuli Piippo <samuli.piippo@qt.io>
Date: Mon, 19 Aug 2019 15:17:13 +0300
Subject: [PATCH] Fix shadow build

Don't write to source directory when using shadow build.
This stops working with Qt 5.14.0.

Change-Id: Ica83b9c810e2a937aab75af131484f5f91570453
---
 tests/auto/scion/scion.pro | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/tests/auto/scion/scion.pro b/tests/auto/scion/scion.pro
index 6e1f7f4..1ebc624 100644
--- a/tests/auto/scion/scion.pro
+++ b/tests/auto/scion/scion.pro
@@ -89,15 +89,15 @@ for (f,ALLSCXMLS) {
 ALLFILES = $$files($$SCXMLS_DIR/*.*, true)
 for (f,ALLFILES) {
     base = $$relative_path($$f,$$absolute_path($$SCXMLS_DIR))
-    file = $$relative_path($$f, $$absolute_path($$PWD))
+    file = $$relative_path($$f, $$absolute_path($$OUT_PWD))
     qrc += '<file alias="$$base">$$file</file>'
 }
 
 contents = $$inc_list "std::function<QScxmlStateMachine *()> creators[] = {" $$func_list "};"
-write_file("scxml/compiled_tests.h", contents)|error("Aborting.")
+write_file("$$OUT_PWD/scxml/compiled_tests.h", contents)|error("Aborting.")
 
 contents = "const char *testBases[] = {" $$testBases "};"
-write_file("scxml/scion.h", contents)|error("Aborting.")
+write_file("$$OUT_PWD/scxml/scion.h", contents)|error("Aborting.")
 
 contents = '<!DOCTYPE RCC><RCC version=\"1.0\">' '<qresource>' $$qrc '</qresource></RCC>'
-write_file("scion.qrc", contents)|error("Aborting.")
+write_file("$$OUT_PWD/scion.qrc", contents)|error("Aborting.")
